OpenSSL/SSL.py

```diff
@@ -1550,6 +1646,16 @@ def get_cipher_version(self):
             version =_ffi.string(_lib.SSL_CIPHER_get_version(cipher))
             return version.decode("utf-8")
 
+    def get_next_proto_negotiated(self):
+        """
+        Get the protocol that was negotiated by NPN.
+        """
+        data = _ffi.new("unsigned char **")
+        data_len = _ffi.new("unsigned int *")
+
+        _lib.SSL_get0_next_proto_negotiated(self._ssl, data, data_len)
+
+        return _ffi.buffer(data[0], data_len[0])[:]
```

Comment:
We are (we allocated it, we have to free it). However, it should be freed up automatically because `data` will go out of scope at the bottom of the function (CPython will free it immediately, PyPy will get around to it). That's fine, because `_ffi.buffer()` will have copied what we care about out already.